### ISSUE TITLE
An Apache/MIT rewrite

### ISSUE THREAD

**Issue by nick-konovalchuk** [WongKinYiu/yolov9#82]
Following up on https://github.com/WongKinYiu/yolov9/issues/10, a bunch of people including me is curious if you're interested in putting managerial or/and development efforts into rewriting certain parts of the repo so that the license could be more permissive

**Comment by nick-konovalchuk** [comment]
@Sharpz7 @zeeshan3945

**Comment by waytechpro** [comment]
Thank you for raising this important consideration! Expanding the accessibility of our repository through a more permissive license could truly amplify collaboration and innovation. Count me in for contributing to the effort! Let's gather our insights and skills to make this vision a reality. Don't hesitate to [reach out](https://g.co/kgs/rxxPvSP) and let's get started!

**Comment by UYousafzai** [comment]
Hello,

I can possibly join in on the effort to rewrite the functionality for a more permissive license. the models however will remain with their original license but at least the code will be separated out.

**Comment by Youho99** [comment]
I can possibly help you to rewrite the basecode

**Comment by vlaminckaxel** [comment]
I'm also willing to collab in this effort!

**Comment by vlaminckaxel** [comment]
Has someone the knowledge to make a list of what parts are non-MIT and needs to be rewritten?

**Comment by szmate00** [comment]
I would be also interested in joining efforts regarding this, would be really useful for my application to have a less strict license!

**Comment by alessandrolamberti** [comment]
I'm also willing to collaborate!

**Comment by slm37102** [comment]
i would like to join too

**Comment by Youho99** [comment]
@nick-konovalchuk How do we organize ourselves?

**Comment by UYousafzai** [comment]
@WongKinYiu would you be interested in organising this?

@Youho99 An alternative I am thinking is I can create a small discord, we fork this code and go through it, separate out the code that needs to be reimplemented and just split it among us who are willing to work on it.

**Comment by GrantorShadow** [comment]
> @WongKinYiu would you be interested in organising this?
> 
> @Youho99 An alternative I am thinking is I can create a small discord, we fork this code and go through it, separate out the code that needs to be reimplemented and just split it among us who are willing to work on it.

This sounds great! I would love to be a part and contribute code for this!

**Comment by GrantorShadow** [comment]
> @nick-konovalchuk How do we organize ourselves?
> 
> Does somebody have an idea ?

How about a small discord / slack server? would be easier to co-ordinate the development for this.

**Comment by alessandrolamberti** [comment]
> How about a small discord / slack server? would be easier to co-ordinate the development for this.

Discord is a good idea, but we probably need a handful of people (2-3) in charge to organize

**Comment by BDannowitzHudl** [comment]
I can also find time to contribute. Please feel free to cc: me on the discord / slack.

**Comment by UYousafzai** [comment]
I have created the Discord Server, however I haven't had the time to set it up properly, I have started looking into some core code and will start doing it properly if and when I find the time from my work.

Discord Link: https://discord.gg/XBpjFQxu

**Comment by Youho99** [comment]
@WongKinYiu 
It would be great if you could join us, to give us some information regarding the parts to be rewritten

**Comment by WongKinYiu** [comment]
Okay

I create a new repo for mit rewrite: https://github.com/WongKinYiu/yolov9mit
Maybe move the discussion to the repo.

I will organize the  code's prototype in 1\~2 weeks.
I think I can handle most of implementation of architectures and loss functions.
And need someone to give great help about dataloader and ddp training.

**Comment by UYousafzai** [comment]
This is perfect.

**Comment by UYousafzai** [comment]
I can pitch in on whatever you need help with, how do you plan on managing the tasks?

**Comment by Sharpz7** [comment]
Hey Folks, sorry I have just seen this.

Great to see this coming together. My bandwidth is limited right now but I look forward to watching this grow :))

**Comment by Sharpz7** [comment]
Please see; https://github.com/WongKinYiu/yolov9mit/issues/3

**Comment by marjankhan945** [comment]
Dear everyone, if Mr. Wong Kin Yiu's introduced blocks for YOLOv9 are moved to YOLOv9mit repository, will there be licensing issue as these codes have already been licensed under GPL3.0?
Or there will be no problem since the inventor agreed to provide a more permissive license to His works?

**Comment by marjankhan945** [comment]
> Dear everyone, if Mr. Wong Kin Yiu's introduced blocks for YOLOv9 are moved to YOLOv9mit repository, will there be licensing issue as these codes have already been licensed under GPL3.0? Or there will be no problem since the inventor agreed to provide a more permissive license to His works?

@Sharpz7 @nick-konovalchuk could you check this? @WongKinYiu could you please also check at your convenience?

**Comment by GrantorShadow** [comment]
Following this!
